libs/coin-polkadot/src/bridge/prepareTransaction.test.ts

```diff
@@ -0,0 +1,48 @@
+import BigNumber from "bignumber.js";
+import { createFixtureAccount, createFixtureTransaction } from "../types/model.fixture";
+import prepareTransaction from "./prepareTransaction";
+import { faker } from "@faker-js/faker";
+
+const mockGetEstimatedFees = jest.fn();
+jest.mock("../logic", () => ({
+  getEstimatedFees: () => mockGetEstimatedFees(),
+}));
+
+describe("prepareTransaction", () => {
+  afterEach(() => {
+    mockGetEstimatedFees.mockClear();
+  });
+
+  it("returns a new Transaction with new fees", async () => {
+    // Given
+    const fees = new BigNumber(faker.number.int(50));
```

Comment:
I don't think using random values in unit tests is the right way... but it's my opinion

In your test specific case I don't think it can hurt but it's a new pattern in our codebase, and has a benefit/ratio cost to discuss

Thoughts:
- Each run is different, we don't provide static values but instead ranges. It goes against a deterministic test and overall against F.I.R.S.T. principle of testing
- We introduce "maybe" more accurate real data (dev has to know and define the real range of values, which changes in real life anyway if we are dealing with a backend => not unit tests role) at the cost of "possible" flakiness. Failing cases are supposed to be tested anyway
- Reduced performance for unit tests runs (random values computing + added library)
- Random values should be used for "exploratory testing" only IMO
- Sometimes there are relations between attributes that faker usage needs to take into account, and it's extra work (imagine relation between spendableBalance, balance, lockedBalance)
- Unit tests are sort of guidelines to the future developer on what to care about, if we really want to add different cases we can use `it.each`

Reply:
My opinion is:
* We don't need or want to always use faker
* Using it prove that our code is indenpendent from this parameter OR that our code is able to manage different value (like you said range)

If you need consistency or have a strong need in relationship between values, you can set your own value instead of faker's one.

I disagree with the performance argument as we are only creating test data, we are losing much more time in E2E test.